x/auth/signing/verify.go

```diff
@@ -57,6 +58,26 @@ func internalSignModeToAPI(mode signing.SignMode) (signingv1beta1.SignMode, erro
 	}
 }
 
+// VerifysignatureCache is a cache of verified signatures
+// If the cached item is not found, it will be verified and added to the cache
+func verifySig(signBytes, sig []byte, pubKey cryptotypes.PubKey) bool {
+	bz := string(append(signBytes, sig...))
```

Comment:
I find this a bit wonky. You're taking the sign bytes and the sig, creating a 2 element slice containing both, and then converting to a string? Can we make this a bit more explicit? Perhaps create a dedicated type with a `String()` method or something.

Reply:
the string is there due to it being comparable. []byte isnt comparable in its natural form. For the separate type do are you forseeing 

type sigkey struct {
signbytes []byte
sig [][]byte
}

or something else?

Reply:
Yes, exactly. Something like that. Then you can do `sigKey.String()` to compare. Just a minor nit though.